malloc/malloc.c

```diff
@@ -57,7 +57,12 @@ void mem_copy(void *dest, const void *src, int n)
 */
 void *resize_memory(void *ptr, int old_size, int new_size)
 {
-
+    char *old_ptr = (char*)ptr;
+    char *temp = malloc(new_size * sizeof(char));
+    for (int i = 0; i < old_size; i++) {
+        temp[i] = old_ptr[i];
+    }
+    return temp;
```

Comment:
once temp is created, ptr becomes obsolete, so you should free up the memory space for ptr before the return line